eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPublisher.scala

```diff
@@ -327,17 +327,14 @@ private class ReplaceableTxPublisher(nodeParams: NodeParams,
 
   // Clean up the failed transaction attempt. Once that's done, go back to the waiting state with the new transaction.
   def cleanUpFailedTxAndWait(failedTx: Transaction, mempoolTx: FundedTx): Behavior[Command] = {
-    context.pipeToSelf(bitcoinClient.abandonTransaction(failedTx.txid))(_ => UnlockUtxos)
```

Comment:
Just to make sure I understand: there is no need to `abandon` htlc txs, as they always compete at the "child" level (we won't publish unless the commitment tx is confirmed). We only need to clean up anchor txs, but that's now done in `Channel`.

It is a _little_ bit weird that the creation and clean up happen at a different place. I guess you didn't want to introduce an adherence to `ClaimLocalAnchorOutputTx`/`HtlcTx`/`ClaimHtlcTx` like in `ReplaceableTxFunder`?

Reply:
> Just to make sure I understand: there is no need to abandon htlc txs, as they always compete at the "child" level (we won't publish unless the commitment tx is confirmed). We only need to clean up anchor txs, but that's now done in Channel.

Yes that's exactly it, that's because we only publish those once the commitment is confirmed. The only transactions we publish before the commit is confirmed are anchor txs.

> It is a little bit weird that the creation and clean up happen at a different place. I guess you didn't want to introduce an adherence to ClaimLocalAnchorOutputTx/HtlcTx/ClaimHtlcTx like in ReplaceableTxFunder?

It's rather that it's hard to clean it up in the tx publishers, because we don't have a reliable enough signal that a commitment transaction confirmed like we do in `Channel.scala`. Especially since we need to distinguish whether it's an alternative commit, or which current one, and also need to abandon other types of transactions (splice txs in the alternative commit case).